sdjournal/journal.go

```diff
@@ -879,6 +879,8 @@ func (j *Journal) TestCursor(cursor string) error {
 
 	if r < 0 {
 		return fmt.Errorf("failed to test to cursor %q: %d", cursor, syscall.Errno(-r))
+	} else if r == 0 {
+		return fmt.Errorf("Cursor parameter is not the same as current position")
```

Comment:
Can you please change this into a top-level error constant and compare against it in the test?